### PR TITLE
artifact_package resource

### PR DESCRIPTION
**Pull request by ivey**

### CODE REVIEW

File: providers/package.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    download_retries new_resource.download_retries
+  end
+
+  package new_resource.name do
+    source tempfile
+    action :install
+  end
+
+  file tempfile do
+    action :delete
```
Comment (KAllan357) [inline providers/package.rb]: Will this mean that every time this resource runs (if the resource itself is not protected by some logic) that wherever we use this resource, we'll be downloading the artifact and deleting it, even if we don't install it again?

Comment (ivey) [inline providers/package.rb]: Addressed.

**Comment by KAllan357** [comment]
:+1:

**Comment by ivey** [comment]
:horse:

**Comment by capoferro** [comment]
:+1:

**Pull request merged**